src/v/ssx/future-util.h

```diff
@@ -428,4 +430,14 @@ seastar::future<T...> with_timeout_abortable(
     return result;
 }
 
+// Create a ready future with template deduction.
+//
+// In most cases you should not need specify a template parameter using this
+// function over seastar's make_ready_future function.
+template<typename T>
+seastar::future<std::remove_cvref_t<T>> now(T&& v) noexcept {
```

Comment:
Otherwise something like: `ssx::now(str)` would have a return type of `ss::future<ss::string&>` which is dangerous.

See: https://godbolt.org/z/xxaoEsTbe

Reply:
Also see: https://github.com/scylladb/seastar/pull/2065

Reply:
> ss::future<ss::string&>

that would just be part of the fun lol

Reply:
I'm glad you have fun tracking down crashes 😉 

Anyways, do you think it's worth changing something here? I am open to suggestions.